npm/translatePolicy.go

```diff
@@ -113,17 +114,44 @@ func craftPartialIptEntrySpecFromOpsAndLabels(ns string, ops, labels []string, s
 	}
 
 	for i, _ := range ops {
+		// TODO need to change this logic, create a list of lsts here and have a single match against it
 		spec = append(spec, craftPartialIptEntrySpecFromOpAndLabel(ops[i], labels[i], srcOrDstFlag, isNamespaceSelector)...)
 	}
 
 	return spec
 }
 
 // craftPartialIptEntrySpecFromSelector :- ns must be "" for namespace selectors
-func craftPartialIptEntrySpecFromSelector(ns string, selector *metav1.LabelSelector, srcOrDstFlag string, isNamespaceSelector bool) []string {
-	labelsWithOps, _, _ := parseSelector(selector)
+// TODO check all references of this func
+func craftPartialIptEntrySpecFromSelector(ns string, selector *metav1.LabelSelector, srcOrDstFlag string, isNamespaceSelector bool) ([]string, []string, map[string][]string) {
+	// parse the sector into labels and maps of multiVal match Exprs
+	labelsWithOps, nsLabelListKVs := parseSelector(selector)
 	ops, labels := GetOperatorsAndLabels(labelsWithOps)
-	return craftPartialIptEntrySpecFromOpsAndLabels(ns, ops, labels, srcOrDstFlag, isNamespaceSelector)
+
+	valueLabels := []string{}
+	listLabelsWithMembers := make(map[string][]string)
+	labelsForSpec := labels
+	// TODO add comments very confusing for sure
```

Comment:
Can you add the detailed comments in this PR? It is good to have comments now when you have contexts in details and helps reviwers to understand the codes as well.